### DISCUSSION TITLE
Option to check syntax only

### DISCUSSION THREAD

**Discussion by vporton** [dfinity/sdk#3380] [Feature Requests]
DFX should support quick syntax/semantics checking of source code. Checking syntax by running `dfx build` is far too slow (apparently, because the compiler does optimizations).

`moc` already has `--check` option, it remains to integrate it into `dfx` like

`dfx check [canister]`

It is questionable, whether all dependent canisters should also be type-checked.

**Comment by viviveevee** [comment]
We collect feature requests over here now and use upvotes to prioritize our work: https://dx.internetcomputer.org/topic/196
